src/compute/src/logging/initialize.rs

```diff
@@ -230,3 +232,17 @@ impl<A: Allocate + 'static> LoggingContext<'_, A> {
         )
     }
 }
+
+#[inline]
+fn downcast_timestamp(time: &dyn ProgressEventTimestamp) -> Option<Timestamp> {
```

Comment:
Maybe call the function `try_downcast_timestamp`? Also, add documentation.

Reply:
Done. Added some documentation, is that sufficient?